### PR TITLE
Fix dsp and yamlconf test

### PR DESCRIPTION
**Pull request by fffw**

### CODE REVIEW

File: src/github.com/getlantern/yamlconf/yamlconf_test.go [hunk 1]
```diff
@@ -214,69 +213,6 @@ func TestCustomPoll(t *testing.T) {
 	}, updated, "Custom polled config should contain correct data")
 }
 
-func TestConfigServer(t *testing.T) {
```
Comment (fffw) [inline src/github.com/getlantern/yamlconf/yamlconf_test.go]: The code this test targets was removed at https://github.com/getlantern/lantern/pull/2989

File: src/github.com/getlantern/peerscanner/dsp/dsp_test.go [hunk 2]
```diff
@@ -39,6 +40,8 @@ func TestRegister(t *testing.T) {
 		assert.NotNil(t, rec, "A new record should have been returned")
 	}
 
+	// give dnssimple enough time to complete the operation
+	time.Sleep(100 * time.Millisecond)
```
Comment (uaalto) [inline src/github.com/getlantern/peerscanner/dsp/dsp_test.go]: I see that you've tried with 100 millis and 1 sec. In order to make it resilient, could we somehow detect success on the DNSSimple side to continue the test? Or maybe don't consider it failed if the fail was on DNSSimple side?

Comment (fffw) [inline src/github.com/getlantern/peerscanner/dsp/dsp_test.go]: Ha I forgot to try run the test locally before submit code. Line 40 already verified that a new record is created, 100 millis is for DNSimple to sync internal state which should be very short I think.

**Comment by uaalto** [comment]
LGTM! Thanks @fffw, having clean tests finally!

**Pull request merged**